pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1023,7 +1024,7 @@ func (v ProjectsVal) Set(projects string) error {
 	if !v.AcceptMultipleProjects && len(prj) > 1 {
 		return fmt.Errorf("multiple GCE projects not supported for command")
 	}
-	providerInstance.Projects = prj
```

Comment:
Same as in `aws`, except there are a few more references to `providerInstance` in this case.

Reply:
Done. Removing the static `providerInstance`s required more refactoring:

1. ConfigureClusterCleanupFlags was moved to the Provider interface instead of the ProviderOpts. My thought here is ProviderOpts are per cluster options, which doesn't really apply to the GC flags.
2. Some options that were defined on the Provider, like the IAM account configured on the machine, were moved to the ProviderOps so that ConfigureClusterFlags could access them.
3. I introduced the InfraProvider interface to expose GCP specific features, since they were previously depending on access to the static `providerInstance`.